pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go

```diff
@@ -275,3 +279,13 @@ func (c *dnsRecord) SetRecordType(recordType extensionsv1alpha1.DNSRecordType) {
 func (c *dnsRecord) SetValues(values []string) {
 	c.values.Values = values
 }
+
+func (c *dnsRecord) ValuesHaveChanged() bool {
+	return c.values.DNSName != c.dnsRecord.Spec.Name ||
+		!pointer.StringEqual(c.values.Zone, c.dnsRecord.Spec.Zone) ||
+		c.values.RecordType != c.dnsRecord.Spec.RecordType ||
+		!pointer.Int64Equal(c.values.TTL, c.dnsRecord.Spec.TTL) ||
+		!reflect.DeepEqual(c.values.Values, c.dnsRecord.Spec.Values) ||
+		c.secret.Name != c.dnsRecord.Spec.SecretRef.Name ||
+		c.values.Type != c.dnsRecord.Spec.Type
```

Comment:
* I am a bit torn if we should really check for changes to all these fields, or only for the ones that are not treated immutable by the `DNSRecord` validation (`Spec.Type`, `Spec.Name`, and `Spec.RecordType` are treated as immutable). If `gardenlet` attempts to change an immutable field, this will be denied by the seed admission controller (and it's considered a bug in `gardenlet`). If we ever find ourselves in a situation where this should be allowed, we would anyway need to do a code change. So I would prefer not to check for changes to immutable fields, it's more consistent with the rest of the code.
* Nit: Could these checks follow some existing ordering, e.g. of the fields in `Values`? Currently it looks pretty randomly ordered.
* I really like `pointer.StringEqual`, we should use this more.

Reply:
done.